apps/aehttp/test/aehttp_contracts_SUITE.erl

```diff
@@ -0,0 +1,1234 @@
+		 acc_c => #{pub_key => CPubkey,
+			    priv_key => CPrivkey,
+			    start_amt => StartAmt},
+		 acc_d => #{pub_key => DPubkey,
+			    priv_key => DPrivkey,
+			    start_amt => StartAmt}},
+    [{accounts,Accounts},{node_name,NodeName}|Config];
+init_per_group(_Group, Config) ->
+    NodeName = aecore_suite_utils:node_name(?NODE),
+    aecore_suite_utils:start_node(?NODE, Config),
+    aecore_suite_utils:connect(NodeName),
+    ToMine = aecore_suite_utils:latest_fork_height(),
+    ct:pal("ToMine ~p\n", [ToMine]),
+    aecore_suite_utils:mine_blocks(NodeName, ToMine),
+    [{node_name,NodeName}|Config].
+
+end_per_group(all_endpoints, _Config) ->
+    ok;
+end_per_group(_Group, Config) ->
+    RpcFun = fun(M, F, A) -> rpc(?NODE, M, F, A) end,
+    {ok, DbCfg} = aecore_suite_utils:get_node_db_config(RpcFun),
+    aecore_suite_utils:stop_node(?NODE, Config),
+    aecore_suite_utils:delete_node_db_if_persisted(DbCfg),
+    ok.
+
+init_per_testcase(_Case, Config) ->
+    [{tc_start, os:timestamp()}|Config].
+
+end_per_testcase(_Case, Config) ->
+    Ts0 = ?config(tc_start, Config),
+    ct:log("Events during TC: ~p", [[{N, aecore_suite_utils:all_events_since(N, Ts0)}
+                                     || {_,N} <- ?config(nodes, Config)]]),
+    ok.
+
+%% ============================================================
+%% Test cases
+%% ============================================================
+
+%% null(Config)
+%%  Does nothing and always succeeds.
+
+null(_Config) ->
+    ok.
+
+%% spending_1(Config)
+%%  A simple test of tokens from acc_a to acc_b.
+
+spending_1(Config) ->
+    NodeName = proplists:get_value(node_name, Config),
+    %% Get account information.
+    #{acc_a := #{pub_key := APubkey,
+		 priv_key := APrivkey},
+      acc_b := #{pub_key := BPubkey}} = proplists:get_value(accounts, Config),
+
+    ct:pal("Top 1 ~p\n", [get_top()]),
+
+    %% Check initial balances.
+    ABal0 = get_balance(APubkey),
+    BBal0 = get_balance(BPubkey),
+    ct:pal("Balances 0: ~p, ~p\n", [ABal0,BBal0]),
+
+    %% Add tokens to both accounts.
+    {ok,200,_} = post_spend_tx(APubkey, 500, 1),
+    {ok,200,_} = post_spend_tx(BPubkey, 500, 1),
+    aecore_suite_utils:mine_blocks(NodeName, 2),
+
+    %% Get balances after mining.
+    ABal1 = get_balance(APubkey),
+    BBal1 = get_balance(BPubkey),
+    ct:pal("Balances 1: ~p, ~p\n", [ABal1,BBal1]),
+
+    %% Transfer money from Alice to Bert.
+    TxHash = spend_tokens(APubkey, APrivkey, BPubkey, 200, 5),
+    MineUntil = fun () -> tx_in_chain(TxHash) end,
+    aecore_suite_utils:mine_blocks_until(NodeName, MineUntil, 10),
+
+    ct:pal("Top 2 ~p\n", [get_top()]),
+
+    %% Check that tx has succeeded.
+    ?assert(tx_in_chain(TxHash)),
+
+    %% Check balances after sending.
+    ABal2 = get_balance(APubkey),
+    BBal2 = get_balance(BPubkey),
+    ct:pal("Balances 2: ~p, ~p\n", [ABal2,BBal2]),
+
+    %% Check that the balances are correct, don't forget the fee.
+    ABal2 = ABal1 - 200 - 5,
```

Comment:
This line is subject to intermittent test failure. This line assumes that `ABal1` reflects [the tx adding 500 tokens](https://github.com/aeternity/epoch/pull/1344/files#diff-c661915907572e4a612ffbdd1c98696eR183) though that is not guaranteed, as that tx may have been included only [while mining for the A->B transaction](https://github.com/aeternity/epoch/pull/1344/files#diff-c661915907572e4a612ffbdd1c98696eR195).

Reply:
There are now relevant tests with the balances checking that they have the correct values.